### ISSUE TITLE
Unpkg "latest" may be a prior major

### ISSUE THREAD

**Issue by lannonbr** [lannonbr/esm-checker#6]
When fetching the package.json for each package, it may fetch an "old version" if the latest `npm publish` was backporting a security bugfix into prior majors. This happened today with [engine.io](https://github.com/socketio/engine.io/commit/f3b761dc560e6cd045ef42939dc28d7138f245e9) today as they pushed version `6.1.1`, `4.1.2` and `5.2.1` in that order, so it thought `5.2.1` was the latest and therefore an audit event was created that engine.io got rid of the `exports.require` field, which technically is incorrect.

I think to fix this, I want to save a "highest version number" for each package when there are updates, so prior to today, engine.io could have had the attribute be `6.1.0` and then I could do some logic to check when I grab the "latest" if it isn't the same major then do some logic to refetch higher versions until I confirm that I found the actual "latest" version. Given most packages of the 2,000 being tracked likely don't update too frequently, this should only add a conditional block within the fetch logic and only add additional http requests infrequently as well.

**Comment by lannonbr** [comment]
Doing some research, The NPM registry will list all versions if you ping `https://registry.npmjs.org/<package_name>` and then collect all of the keys under the `versions` field. Although these files are quite large for more popular files (ex: webpack is >1MB compressed, ~10MB uncompressed), the low rate of needing to fetch these should not cause any performance issues.

**Comment by lannonbr** [comment]
After doing further exploration on this I am thinking of just scrapping this feature due to too much technical debt that could pile up due to adding this.

At the time of writing this, there are some various ways people are using `latest` differently:

- Formidable & Vue have `2.x` branches as their stable but `3.x` branches are available. So if I fetch "greatest semver", those will not be aligned.
- rc-select and html-minifier-terser have alpha builds on the `latest` branch so doing a command like `npm install rc-select` would result in you downloading explicit alpha software which people may not be fond of.
- jsprim has a 2.x branch which was stable, but they pushed a 1.4.x patch release so you wouldn't know of a 2.x line unless you dug into the package as I described in the original post in this thread.

Due to all of these gotchas, I could technically add a watchlist of packages that have these caviats and how to have this project handle them, but due to the low number of these types of packages, it feels like too much extra work to maintain in exchange for having false-positives very infrequently.

Closing for now. I may reopen this if I find a better solution that doesn't add something I have to manually maintain, but for the time being that doesn't exist so I will close it
